### PR TITLE
Document customizing import schema

### LINKED ISSUE #202: More information from the Upload file
Ingestion of primary tumor data fields (ICD-O topography code and label, ICD-O morphology code and preferred histology term, and ICD-O behavior code) from the DeID Upload File.

This should be a modification of the schema.

### PR DESCRIPTION
**Pull request by naglepuff**
Closes #202 

This PR includes updates to the ``CUSTOMIZING.rst`` documentation with details on modifying or extending the import schema. It documents how new columns can be added to the schema file to allow for changes to import spreadsheets. 

This PR also includes an example schema file, ``wsi_deid/schema/importManifestSchema.example.json`` to further illustrate how the schema can be modified. The example schema adds three new fields, and removes a field from the "required" property of the schema, to show some examples of changes.

### CODE REVIEW

**Comment by naglepuff** [comment]
Is this sufficient for #202? Since this is mostly documentation changes I was planning on creating another branch/PR to tackle #237

File: docs/CUSTOMIZING.rst [hunk 1]
```diff
@@ -72,3 +72,10 @@ Import and Export Controls
...
+
+Import Schema Modification
+--------------------------
+
+Import excel files can be customized, allowing for additional metadata to be captured during the import process. The expected schema for import files is described in ``wsi_deid/schema/importManifestScheme.json``. An example of a modified schema can be found in ``importManifestSchema.example.json``. In this example schema, the fields  ``Tumor_Rec_Number``, ``Histology_Code``, and ``Behavior_Code`` have been added. Each of these new fields uses a regular expression for validation. Enums (a set of values) can be used for validation instead of pattern matchin (see the property ``Proc_Type`` for an example).
```
Comment (manthey) [inline docs/CUSTOMIZING.rst:79]: matchin -> matching

**Comment by manthey** [comment]
Perfect.  Definitely #237 can be a different PR.

**Comment by manthey** [comment]
I think the one additional piece of documentation would be to comment on adding a volume mount to the docker-compose file to replace the schema with the alternate schema (or, instead, copy the schema in via the Dockerfile so that, if replaced, it gets used).

**Comment by naglepuff** [comment]
> I think the one additional piece of documentation would be to comment on adding a volume mount to the docker-compose file to replace the schema with the alternate schema (or, instead, copy the schema in via the Dockerfile so that, if replaced, it gets used).

So I did some more testing with mounting a volume and I'm not so sure it will work. 

In my `docker-compose.local.yml` file, I'm specifying the `dsarchive/wsi_deid:stable` image. I've also added a volume mount for the schema file, like this `- ../../wsi_deid/schema/importManifestSchema.json:/wsi_deid/wsi_deid/schema/importManifestSchema.json`

After running `docker-compose (-f ...) up`, the locally defined schema is in the correct location in the container, so it was mounted to that location successfully. However, if trying to use an import spreadsheet with new columns defined in the local schema, the import fails.

I believe this is because one the container is running, the `wsi_deid` plugin is running from `/usr/local/lib/python3.9/dist-packages/wsi_deid`. When the import process goes to find the schema to use for validation it gets a schema from `/usr/local/lib/python3.9/dist-packages/wsi_deid/schema/importManifestSchema.json`. I'm not sure how we'd mount a schema file in the correct location without having all deployments build docker images locally and using the `:latest` image or changing where the import logic looks for the file.

**Comment by manthey** [comment]
> So I did some more testing with mounting a volume and I'm not so sure it will work.
> 
> In my `docker-compose.local.yml` file, I'm specifying the `dsarchive/wsi_deid:stable` image. I've also added a volume mount for the schema file, like this `- ../../wsi_deid/schema/importManifestSchema.json:/wsi_deid/wsi_deid/schema/importManifestSchema.json`

Ah, we aren't installing wsi_deid in editable mode.  There are two alternative ways to handle this.

The first is to have the second half of the mount position be where it is in the installed python (e.g., `/usr/local/lib/python3.9/dist-packages/wsi_deid/schema/importManifestSchema.json`).

The second way would be to change https://github.com/DigitalSlideArchive/DSA-WSI-DeID/blob/master/Dockerfile#L55 to have `-e` in the command.  In looking at that line, we aren't using `girder[mount]`, so we could remove it from the install command (but if we left it there, it should be installed without the `-e` option, so we'd split the install into two commands).

**Comment by naglepuff** [comment]
> The first is to have the second half of the mount position be where it is in the installed python (e.g., `/usr/local/lib/python3.9/dist-packages/wsi_deid/schema/importManifestSchema.json`).
> 
> The second way would be to change [`master`/Dockerfile#L55](https://github.com/DigitalSlideArchive/DSA-WSI-DeID/blob/master/Dockerfile?rgh-link-date=2021-11-08T20%3A40%3A29Z#L55) to have `-e` in the command. In looking at that line, we aren't using `girder[mount]`, so we could remove it from the install command (but if we left it there, it should be installed without the `-e` option, so we'd split the install into two commands).

I ended up going with the first option (20429b9), since I could test it immediately to verify that it works. If you'd prefer the second way. Please let me know.

**Review by manthey — APPROVED** [review]

**Pull request merged**